src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,473 @@
+
+            if self.cover == "voronoi":
+                assert self.complex_type is not "nerve"
+                self.complex.set_cover_from_Voronoi(self.voronoi_samples)
+
+            elif self.cover == "functional":
+
+                if self.complex_type == "gic":
+
+                    self.complex.set_function_from_range(self.filters)
+
+                    if self.resolutions is None:
+                        self.complex.set_automatic_resolution()
+                    else:
+                        self.complex.set_resolution_with_interval_number(self.resolutions[0])
+
+                    if self.gains is None:
+                        self.complex.set_gain(.33)
+                    else:
+                        self.complex.set_gain(self.gains[0])
+
+                    self.complex.set_cover_from_function()
+
+                elif self.complex_type == "nerve": # it's actually better to use Mapper with constant clustering
+                    self.complex = _MapperComplex(filters=self.filters, filter_bnds=self.filter_bnds, colors=self.colors, 
+                                                 resolutions=self.resolutions, gains=self.gains, inp=self.input_type, 
+                                                 clustering=self._constant_clustering, mask=self.mask, N=self.N, beta=self.beta, C=self.C)
+                    self.complex.fit(X)
+                    self.simplex_tree = self.complex.mapper_
+                    self.node_info = self.complex.node_info_
+
+            elif self.cover == "precomputed":
+                self.complex.set_cover_from_range(self.assignments)
+
+            if self.complex_type is not "nerve" or self.cover is not "functional":
+
+                self.complex.set_mask(self.mask)
+
+                self.complex.find_simplices()
+                simplex_tree = self.complex.create_simplex_tree()
+            
+                self.simplex_tree = SimplexTree()
+                idv, names = 0, {}
+                for v,_ in simplex_tree.get_skeleton(0):
+                    if len(self.complex.subpopulation(v[0])) > self.mask:
+                        names[v[0]] = idv
+                        self.simplex_tree.insert([idv])
+                        idv += 1
+                for s,_ in simplex_tree.get_filtration():
+                    if len(s) >= 2 and np.all([len(self.complex.subpopulation(v)) > self.mask for v in s]):
+                        self.simplex_tree.insert([names[v] for v in s])
+                self.node_info = {}
+                for v,_ in simplex_tree.get_skeleton(0):
+                    if len(self.complex.subpopulation(v[0])) > self.mask:
+                        node = names[v[0]]
+                        pop = self.complex.subpopulation(v[0])
+                        self.node_info[node] = {"indices": pop, "size": len(pop), "colors": [self.complex.subcolor(v[0])]}
+    
+
+        elif self.complex_type == "mapper":
+
+            assert self.cover is not "voronoi"
```

Comment:
`is not` should only be used for `None`, for instance. Use `!=` operator for literals.
```suggestion
            assert self.cover != "voronoi"
```

Reply:
Thanks! Done.